lib/telemetry_metrics.ex

```diff
@@ -9,7 +9,9 @@ defmodule Membrane.TelemetryMetrics do
 
   @type label() :: list()
 
-  @dialyzer [{:nowarn_function, emit_event?: 1}]
+  # Suppresses Dialyzer false-positive error occurring due to fact, that function contains match on module attribute known at compile time but can be overridden in the config file.
+  # Dialyzer returns an error saying, that pattern can never match the type, but it might be not true, in the case when somebody will change its value in the custom config file.
+  @dialyzer [{:no_match, emit_event?: 1}]
```

Comment:
This description is way too verbose, but it can be removed altogether by simply fixing the error. Instead of defining a condition inside helper function, conditionally define the function and it seems to work, ensure it won't generate warnings in code using the macros here:
```elixir
  cond do
    not @enabled -> defp emit_event?(_event), do: false
    @events == :all -> defp emit_event?(_event), do: true
    is_list(@events) -> defp emit_event?(event), do: event in @events
    true -> defp emit_event?(_event), do: false
  end
```